### PR TITLE
add doc for message missing-final-newline / C0304

### PR DESCRIPTION
**Pull request by hofrob**
## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :scroll: Docs          |

## Description

Add doc for `missing-final-newline / C0304`.

Refs #5953

### CODE REVIEW

**Comment by hofrob** [comment]
FYI: I saw other merged PRs related to #5953 without a towncrier news fragment, so I assumed it's not necessary here.

Regarding this PR: It's not possible to submit a "bad example" that triggers a pylint error since the pre commit hook will fix the missing newline.

All the other CI checks should be fine.

**Review by Pierre-Sassoulas — CHANGES_REQUESTED** [review]
Thank you for opening a MR :) You'll need to add an exclude for bad.py in the pre-commit hook that fix end of file here: https://github.com/PyCQA/pylint/blob/main/.pre-commit-config.yaml#L15

File: doc/data/messages/m/missing-final-newline/bad.py [hunk 1]
```diff
@@ -0,0 +1,9 @@
+# using LF
+
+foo(bar, baz)  # \n
```
Comment (Pierre-Sassoulas) [inline doc/data/messages/m/missing-final-newline/bad.py:3]:
```suggestion
eat("apple", "candy") # \n
```

File: doc/data/messages/m/missing-final-newline/bad.py [hunk 2]
```diff
@@ -0,0 +1,9 @@
...
+print(123)  # EOF
+
+# using CRLF
+
+foo(bar, baz)  # \r\n
```
Comment (Pierre-Sassoulas) [inline doc/data/messages/m/missing-final-newline/bad.py:8]:
```suggestion
eat("apple", "candy")  # \r\n
```

File: doc/data/messages/m/missing-final-newline/good.py [hunk 3]
```diff
@@ -1 +1,9 @@
-# This is a placeholder for correct code for this message.
+# using LF
+
+foo(bar, baz) #  \n
```
Comment (Pierre-Sassoulas) [inline doc/data/messages/m/missing-final-newline/good.py:3]:
```suggestion
eat("apple", "candy") #  \n
```

File: doc/data/messages/m/missing-final-newline/good.py [hunk 4]
```diff
@@ -1 +1,9 @@
...
+print(123)  # \nEOF
+
+# using CRLF
+
+foo(bar, baz)  # \r\n
```
Comment (Pierre-Sassoulas) [inline doc/data/messages/m/missing-final-newline/good.py:8]:
```suggestion
eat("apple", "candy")  # \r\n
```

**Review by Pierre-Sassoulas — CHANGES_REQUESTED** [review]
Do not use the taboo words that makes all the beginner disappear 😄 Otherwise 👌

**Review by Pierre-Sassoulas — APPROVED** [review]

File: doc/data/messages/m/missing-final-newline/good.py (hunk 4 above)
Comment (hofrob) [reply doc/data/messages/m/missing-final-newline/good.py]: Oh, yes that makes sense! Should I replace `print(123)` too?

Comment (Pierre-Sassoulas) [reply doc/data/messages/m/missing-final-newline/good.py]: No it's ok, school children generally knows about numbers under 4 😛

**Review by Pierre-Sassoulas — APPROVED** [review]

File: doc/data/messages/m/missing-final-newline/bad.py [hunk 5]
```diff
@@ -0,0 +1,9 @@
+# using LF
+
+eat("apple", "candy")  # \n
+print(123)  # EOF
+
+# using CRLF
+
+eat("apple", "candy")  # \r\n
+print(123)  # EOF
```
Comment (Pierre-Sassoulas) [inline doc/data/messages/m/missing-final-newline/bad.py:9]:
```suggestion
print(123)  # EOF
# +1: [missing-final-newline]
```

**Review by Pierre-Sassoulas — CHANGES_REQUESTED** [review]
You can test locally with ``pytest doc -k missing-final``

File: doc/data/messages/m/missing-final-newline/bad.py (hunk 5 above)
Comment (hofrob) [reply doc/data/messages/m/missing-final-newline/bad.py]: Thanks! It worked without a line offset.

**Review by Pierre-Sassoulas — APPROVED** [review]

**Comment by Pierre-Sassoulas** [comment]
Thank you 🎉

**Pull request merged**